language/go/fileinfo.go

```diff
@@ -569,8 +569,18 @@ func checkConstraints(c *config.Config, os, arch, osSuffix, archSuffix string, t
 	}
 
 	goConf := getGoConfig(c)
-	checker := func(tag string, ts tagSet) bool {
-		if isIgnoredTag(tag) {
+
+	isIgnoredTag := func(tag string) bool {
+		return goConf.genericTags[tag]
+	}
+
+	if tags != nil && tags.expr != nil {
+		replacedExpr, _ := dropNegationForIgnoredTags(tags.expr, isIgnoredTag)
+		tags, _ = newBuildTags(replacedExpr)
```

Comment:
Handle any errors returned.

It looks like these functions only return errors if there's some unknown syntax. It would be fine to change that to a panic I think.

Reply:
This is one of the main reasons I had marked it as draft, I thought this didn't look right but I also didn't want to change the function's return value to bubble up errors. If we're fine with the way this is implemented by panicking here I'll do that and mark the PR as reviewable then.